src/sif/dynamiccost.cc

```diff
@@ -489,16 +489,16 @@ void ParseBaseCostOptions(const rapidjson::Value& json,
 void ParseCosting(const rapidjson::Document& doc,
                   const std::string& costing_options_key,
                   Options& options) {
-  // if specified, get the costing options in there
-  for (auto i = Costing::Type_MIN; i <= Costing::Type_MAX; i = Costing::Type(i + 1)) {
+  // get the needed costing options in there
+  for (const auto& costing_type : kCostingTypeMapping.at(options.costing_type())) {
```

Comment:
instead of going over all possible costings, I did a map with the relevant ones per request costing. the value has to be a vector to support multimodal, but most costings map to a single one

Reply:
this function is only being used for valid `options` and we can expect a `option.costing_type` to be defined.